tests/integration/test_migrator.py

```diff
@@ -142,7 +164,41 @@ def test_continuous_migration_handles_multiple_records(app, record_1502655_and_1
     assert expected == result
 
 
+def test_continuous_migration_avoids_jamming(app, many_times_same_record):
+    with app.app_context():
+        with mock.patch.dict(app.config, {'CELERY_ALWAYS_EAGER': False}):
```

Comment:
I just realized one key thing: by the time you're running this line, the Celery app has already been instantiated, and it's too late to modify its behavior. I think that @spirosdelviniotis had a similar issue in the past, no?